test/plots/xyPlotTests.ts

```diff
@@ -2,268 +2,348 @@
+        const xScale = new Plottable.Scales.Linear();
+        const plot = new Plottable.XYPlot<number, number>();
+        plot.x(0, xScale);
+        plot.anchor(svg);
+        plot.computeLayout();
+
+        assert.deepEqual(xScale.range(), [0, plot.width()], "range extends to the width");
+
+        svg.remove();
+      });
+
+      it("extends the y scale range from the component height", () => {
+        const yScale = new Plottable.Scales.Linear();
+        const plot = new Plottable.XYPlot<number, number>();
+        plot.y(0, yScale);
+        plot.anchor(svg);
+        plot.computeLayout();
+
+        assert.deepEqual(yScale.range(), [plot.height(), 0], "range extends from the height");
+
         svg.remove();
       });
 
-      it("automatically adjusts range when a Scale is attached after rendering", () => {
-        plot.x((d: any) => d.a)
-            .y((d: any) => d.b);
+      it("extends the y scale range to the component height if it is a Category scale", () => {
+        const yScale = new Plottable.Scales.Category();
+        const plot = new Plottable.XYPlot<number, string>();
+        plot.y("one", yScale);
+        plot.anchor(svg);
+        plot.computeLayout();
 
-        plot.renderTo(svg);
-        plot.x((d: any) => d.a, xScale)
-            .y((d: any) => d.b, yScale);
+        assert.deepEqual(yScale.range(), [0, plot.height()], "range extends to the height");
 
-        assert.deepEqual(xScale.domain(), [-7, 7], "X domain has been set automatically");
-        assert.deepEqual(xScale.range(), [0, SVG_WIDTH], "X range has been set automatically");
-        assert.deepEqual(yScale.domain(), [-7, 7], "Y domain has been set automatically");
-        assert.deepEqual(yScale.range(), [SVG_HEIGHT, 0], "Y range has been set automatically");
         svg.remove();
       });
     });
 
-    describe("Deferred Rendering", () => {
-      let svg: d3.Selection<void>;
-      let xScale: Plottable.Scales.Linear;
-      let yScale: Plottable.Scales.Linear;
+    describe("managing the x property", () => {
       let plot: Plottable.XYPlot<number, number>;
 
-      let nativeTimeout: Function;
-
       beforeEach(() => {
-        svg = TestMethods.generateSVG(500, 500);
-        xScale = new Plottable.Scales.Linear();
-        yScale = new Plottable.Scales.Linear();
         plot = new Plottable.XYPlot<number, number>();
-        (<any> plot)._createDrawer = (dataset: Plottable.Dataset) => createMockDrawer(dataset);
 
-        // Replacing timeout with instant invocation to avoid waiting for the deferred rendering
-        nativeTimeout = (<any>window).setTimeout;
-        (<any>window).setTimeout = function(fn: Function) {
-          fn.call(this);
-        };
+        // HACKHACK: Must install y scale https://github.com/palantir/plottable/issues/2934
+        plot.y(0, new Plottable.Scales.Linear());
       });
 
       afterEach(() => {
-        (<any>window).setTimeout = nativeTimeout;
+        plot.destroy();
```

Comment:
isn't this the same as calling svg.remove() in afterEach? we can't see plot for failing tests
can you do this for only passed tests?